wellregistry/registry/admin/monitoring_location.py

```diff
@@ -216,11 +217,95 @@ def _has_permission(perm, user, obj=None):
 
     return user.has_perm(perm) and (not obj or obj.agency.agency_cd in _get_groups(user))
 
+def _get_usergroups(request):
+    user_groups = ""
+    if not request.user.is_superuser:
+        user_groups = re.search("\[(.*)\]", str(_get_groups(request.user))).group(1)
+        user_groups = " where agency.agency_cd IN (" + user_groups + ") "
+    return user_groups
+
+def _get_where_clause(request, user_groups):
+    where_clause = ""
+    sep = ""
+    query_params = request.body.decode().split("&")
+    for query_param in query_params:
+        param = query_param.split("=")
+        if param[0] == "_selected_action":
+            where_clause += sep + param[1]
+            sep = ","
+    if len(where_clause) > 0:
+        if user_groups == "":
+            where_clause = " where registry_monitoringlocation.id IN (" + where_clause + ") \n"
+        else:
+            where_clause = user_groups + " and registry_monitoringlocation.id IN (" + where_clause + ") \n"
+    return where_clause
+
+def _get_monitoringlocation_queryset(request):
```

Comment:
Did you try adding the select_related? Just curious what kind of performance boost it provided (should have been some) and if so how did it compare to this?

Reply:
i did try selected_related, among many other things
I could not get this down under 20 seconds.. With .raw() 
and the way I build the query, the download (tested with 2900 records)
is now under a second